File: OOPs/Constructors/ConstructorOverloading.java
public class ConstructorOverloading {
    private double length;
    private double width;

    // Default constructor
    public ConstructorOverloading() {
        length = 1;
        width = 1;
    }

    // Constructor with one parameter
    public ConstructorOverloading(double length) {
        this.length = length;
        this.width = 1;
    }

    // Constructor with two parameters
    public ConstructorOverloading(double length, double width) {
        this.length = length;
        this.width = width;
    }

    public double area() {
        return length * width;
    }

    public static void main(String[] args) {
        ConstructorOverloading rect1 = new ConstructorOverloading();          // Default constructor
        ConstructorOverloading rect2 = new ConstructorOverloading(5);        // Single-parameter constructor
        ConstructorOverloading rect3 = new ConstructorOverloading(4, 6);     // Two-parameter constructor

        System.out.println("Area of rect1: " + rect1.area());  // Output: 1
        System.out.println("Area of rect2: " + rect2.area());  // Output: 5
        System.out.println("Area of rect3: " + rect3.area());  // Output: 24
    }
}

File: OOPs/Constructors/constructor_overloading.cpp
#include <iostream>
using namespace std;

class Rectangle {
private:
    double length, width;

public:
    // Default constructor
    Rectangle() {
        length = 1;
        width = 1;
    }

    // Constructor with one parameter
    Rectangle(double l) {
        length = l;
        width = 1;
    }

    // Constructor with two parameters
    Rectangle(double l, double w) {
        length = l;
        width = w;
    }

    double area() {
        return length * width;
    }
};

int main() {
    Rectangle rect1;          // Calls default constructor
    Rectangle rect2(5);       // Calls single-parameter constructor
    Rectangle rect3(4, 6);    // Calls two-parameter constructor

    cout << "Area of rect1: " << rect1.area() << endl;  // Output: 1
    cout << "Area of rect2: " << rect2.area() << endl;  // Output: 5
    cout << "Area of rect3: " << rect3.area() << endl;  // Output: 24

    return 0;
}
